atr/sbom/utilities.py

```diff
@@ -164,3 +195,35 @@ def _map_severity(severity: str) -> str:
         if sev == "moderate":
             return "medium"
     return "unknown"
+
+
+def _record_task(task: str, revision: str, doc: yyjson.Document, patch_ops: models.patch.Patch) -> models.patch.Patch:
+    properties: list[dict[str, str]] | None = get_pointer(doc, "/properties")
```

Comment:
Are all of these types definitely over `dict[str, str]`?

Reply:
Even tools? Because I don't understand:

```
    tools: dict[str, str] | None = get_pointer(doc, "/metadata/tools")
    services: list[dict[str, str]] | None = get_pointer(doc, "/metadata/tools/services")
```

If services are in tools, and tools is `{str: str}` but service values are themselves `{str: str}`, doesn't this mean that some of the values of tools are `{str: str}`?

Reply:
Rather, that some of the values of tools are `[{str: str}]`, since services is a list. But in any case, not `str`.

Reply:
Yes, you're right - sorry I saw the comment was on properties and didn't look any further!